sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/converter/RubyVisitor.java

```diff
@@ -120,6 +128,61 @@ public IRubyObject onDefs(ThreadContext context, IRubyObject node) {
     return createFunctionDeclaration(context, node);
   }
 
+  @JRubyMethod(name = "on_if")
+  public IRubyObject onIf(ThreadContext context, IRubyObject node) {
+    IRubyObject updatedNode = visitNode(context, "on_if", node);
+
+    Token mainKeyword = noLocation(updatedNode, "keyword") ? null : getMetaData(updatedNode, "keyword").tokens().get(0);
+    if (mainKeyword == null || mainKeyword.text().equals("unless")) {
+      // Ternary operator and "unless" are not considered as "IfTree" for now
+      return createNativeTreeAdapter(updatedNode);
+    }
+
+    Token elseKeyword = noLocation(updatedNode, "else") ? null : getMetaData(updatedNode, "else").tokens().get(0);
```

Comment:
I'm not sure it's a good idea to have shared logic for first `noLocation` used in `process` and this one. There is only one line common, and semantic is different